tests/test_cleanup.py

```diff
@@ -1770,3 +1785,137 @@ def test_not_plugged_retry(self):
         # Assert
         self.assertIsNotNone(sr)
 
+
+class TestLockActive(unittest.TestCase):
+    # We mock flock.MockWriteLock so that we can easily fake
+    # up an lock being held by another process.
+    class MockWriteLock: # pragma: no cover
+        test_case = None
+
+        def __init__(self, fd):
+            self.fd = fd
+            self._held = False
+
+        def is_externally_locked(self):
+            return self.test_case.is_externally_locked(self.fd)
+
+        def lock(self):
+            if self.is_externally_locked():
+                raise AssertionError("Failed attempt to take out lock")
+            self._held = True
+
+        def trylock(self):
+            if self._held:
+                return False
+            if self.is_externally_locked():
+                return False
+            self._held = True
+            return True
+
+        def held(self):
+            return self._held
+
+        def unlock(self):
+            self._held = False
+
+        def test(self):
+            """Returns the PID of the process holding the lock or -1 if the lock
+            is not held."""
+            if self._held:
+                return os.getpid()
+            elif self.is_externally_locked():
+                return 1
+            else:
+                return -1
+
+    def setUp(self):
+        tmp_dir = TemporaryDirectory()
```

Comment:
It doesn't look like we actually use this as a temporary directory anymore just as a source for a named directory to inject into a mock? If so then we can simplify it down to just a statically defined literal path?

Reply:
Lock will be creating directories and lock files in that directory, so I'd rather it were a temporary directory that gets clean up.